### PR TITLE
milestone_4

### PR DESCRIPTION
**Pull request by BrianSammit**
Hello reviewers 👋 😄 
-------------
In this pull request I create: 

- A logic directory
- In this logic directory create `calculate.js` and `operate.js`
- Add `big.js` to the dependencies
- Implement the `calculate.js` module 
- Implement the `operate.js` module
--------------
Thanks in advance 👍 💯 
I'll be waiting for any advice and requests needed.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
+// import calculate from '../logic/calculate';
```
Comment (rindrajosia) [inline src/components/App.js]: `calculate.js` should be imported in `App` component

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      result = numOne.minus(numTwo);
+      break;
+    case 'X':
+      result = numOne.times(numTwo);
+      break;
+    case '÷':
+      result = numOne.div(numTwo);
+      break;
+    case '%':
+      result = numOne.mod(numTwo);
```
Comment (rindrajosia) [inline src/logic/operate.js]: This operation should be: percentage

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  switch (operation) {
+    case '+':
+      result = numOne.plus(numTwo);
+      break;
+    case '-':
+      result = numOne.minus(numTwo);
+      break;
+    case 'X':
+      result = numOne.times(numTwo);
```
Comment (rindrajosia) [inline src/logic/operate.js:16]: This operation should be: multiply

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- calculate.js module is added :+1: 
- `operate.js` module is added :+1: 

To improve :factory_worker: [REQUIRED]:

- [ ] Your readme should be descriptive.  To be descriptive you have to add the title of the project, a description about it in the README . ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [ ] `calculate.js` should be imported in `App` component (Please chack my inline comment :arrow_double_down: )


- [ ]  Inline comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history.
- [ ] It will be nice if the button name correspond to the correct operation  (Please chack my inline comment :arrow_double_down: )

Nice to have [OPTIONAL] :

- [ ] 

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by taiwo2 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 
@BrianSammit  Great job for implementing suggested changes from the previous Code Reviewer, however changes is required
### highlights

- [x] Inline comments 👍🏻 
- [x] calculate.js should be imported in App.js 👍🏻 
- [x] uses `buttonName` argument to mutate calculator data object and return its modified version. I can see for `buttonName` === "+/-", it should multiply total and next by -1 to make it negative or positive. 👍🏻

### Improve

- [ ] kindly make your Getting Started mentioned  npm install in the installation aspect 

[OPTIONAL ] 
- [ ] Kindly implement for other corresponding `buttonName` in your calculate.js file

### once significant changes made kindly request for another review

**Pull request merged**